File: pipeline/0_data_prep/pylib/old/preprocessGBIF.py
import pandas as pd

# note - doesn't work yet, just sketch
# should also merge different record sets
def preprocessMergeRecords(csv_names,id_cols,proj_root="../.."):
    """
        Preprocess and merge records data,
        :param list<str> csv_names: a list of names of csvs within the data folder to merge into a single all_records.csv
        :param list<str> id_cols: a list of columns specifying which respective columns in the csvs to draw record IDs from
        :param str proj_root: the location of the project folder i.e. dragonfly-patterner containing an /R/src/preprocessiNat.R script to source from
    """

    # merge record dataframes
    # create empty dataframe
    df = pd.DataFrame()
    # for every csv
    for name,id_col in tuple(csv_names,id_cols):
        # read in csv
        next_df = pd.read_csv(filepath_or_buffer=proj_root + "/" + name + ".csv") #, header=True, sep="\t",quotechar="")
        df['recordID'] = df[id_col]
        df = pd.concat(df, next_df)

    #df['numImages'] = df['mediaType'].count("StillImage")

    #keeps = ["occurrenceID","family","genus","species","infraspecificEpithet",
    #         "decimalLatitude","decimalLongitude","coordinateUncertaintyInMeters",
    #         "eventDate","day","month","year","taxonKey",
    #         "catalogNumber","identifiedBy","recordedBy","issue","numImages"]
    #df = df[keeps]

    #df = df[df['numImages'] != 0,:]

    df.to_csv(proj_root + "/data/all_records.csv")

File: R/src/old/preprocessGBIF.R

# add useful cols and remove extraneous cols from data downloaded from GBIF (including iNat research grade, AntWeb, or others)
preprocessGBIF <- function(gbif_csv_name,proj_root="../..")
{
  data <- read.csv(paste0(proj_root, "/data/", gbif_csv_name, ".csv"),header=TRUE,row.names=NULL,sep="\t",quote="")
  
  # add column for number of images
  library(stringr)
  data$numImages <- str_count(data$mediaType, "StillImage")
  # data$numImages[data$numImages == 0] <- 1
  print("Added numImages column containing number of images per observation")

  keeps <- c("occurrenceID","family","genus","species","infraspecificEpithet",
             "decimalLatitude","decimalLongitude","coordinateUncertaintyInMeters",
             "eventDate","day","month","year","taxonKey",
             "catalogNumber","identifiedBy","recordedBy","issue","numImages")
  data <- data[keeps]
  
  print("Removed extraneous columns")
  
  data <- data[data$numImages != 0,]
  print("Removed observations without images")
  
  write.csv(data,paste0(proj_root, "/data/", gbif_csv_name, ".csv"))
  print("Wrote updated data .csv to data folder")
}
